backend/src/main/java/com/festago/festival/repository/FestivalFilter.java

```diff
@@ -0,0 +1,40 @@
+package com.festago.festival.repository;
+
+import static com.festago.festival.repository.FestivalSpecification.afterEndDate;
+import static com.festago.festival.repository.FestivalSpecification.afterStartDate;
+import static com.festago.festival.repository.FestivalSpecification.all;
+import static com.festago.festival.repository.FestivalSpecification.beforeEndDate;
+import static com.festago.festival.repository.FestivalSpecification.beforeStartDate;
+
+import com.festago.common.exception.BadRequestException;
+import com.festago.common.exception.ErrorCode;
+import com.festago.festival.domain.Festival;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.function.Function;
+import org.springframework.data.jpa.domain.Specification;
+
+public enum FestivalFilter {
+    ALL(currentTime -> all()),
+    PROGRESS(currentTime -> afterStartDate(currentTime)
+        .and(beforeEndDate(currentTime))),
+    PLANNED(currentTime -> beforeStartDate(currentTime)),
+    END(currentTime -> afterEndDate(currentTime));
+
+    private final Function<LocalDate, Specification<Festival>> filter;
+
+    FestivalFilter(Function<LocalDate, Specification<Festival>> filter) {
+        this.filter = filter;
+    }
+
+    public static FestivalFilter from(String filterName) {
+        return Arrays.stream(FestivalFilter.values())
+            .filter(festivalFilter -> festivalFilter.name().equalsIgnoreCase(filterName))
+            .findAny()
+            .orElseThrow(() -> new BadRequestException(ErrorCode.INVALID_FESTIVAL_FILTER));
+    }
```

Comment:
크게 고려해야 할 상황은 아니지만...
스트림을 매번 돌리기 보단 스위치를 사용하는게 좋아 보이네요.
물론 새로운 Filter가 추가되면 해당 코드를 변경해야 합니다.
```suggestion
public static FestivalFilter from(String filterName) {
        return switch (filterName) {
            case "ALL" -> ALL;
            case "PROGRESS" -> PROGRESS;
            case "PLANNED" -> PLANNED;
            case "END" -> END;
            default -> throw new BadRequestException(ErrorCode.INVALID_FESTIVAL_FILTER);
        };
    }
```

Reply:
저도 연산마다 Stream 이 생성되는 것이 비효율적이라 생각했지만
Java 코드 레벨에서의 비효율이 실질적으로 미치는 영향이 크지 않다 생각했습니다.
하지만 But 한번 적용해보는 것도 좋아보여 반영했습니다!